File: pset6/caesar/caesar.py
from sys import argv


if not len(argv) == 2:
    print("Usage: python caesar.py key")
    exit(1)

key = int(argv[1])

plain = input("plaintext: ")

print("ciphertext: ", end="")
for c in plain:
    if not c.isalpha():
        print(c, end="")
    else:
        if c.isupper():
            c = ord(c)
            c -= 65
            c += key
            c %= 26
            c += 65
        else:
            c = ord(c)
            c -= 97
            c += key
            c %= 26
            c += 97

        print(chr(c), end="")

print()

File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc == 2) 
    {
        for (int i = 0; i < strlen(argv[1]); i++)
        {
            if (argv[1][i] < 48 || argv[1][i] > 57) 
            {
                printf("Usage: ./caesar key\n");
                return 1;
            }
        }
        
        int key = atoi(argv[1]);
        string plaintext = get_string("plaintext: ");
        
        printf("ciphertext: ");
        for (int i = 0; i < strlen(plaintext); i++) 
        {
            if (plaintext[i] >= 'a' && plaintext[i] <= 'z')
            {
                printf("%c", (plaintext[i] + key - 'a') % 26 + 'a');
            }
            else if (plaintext[i] >= 'A' && plaintext[i] <= 'Z')
            {
                printf("%c", (plaintext[i] + key - 'A') % 26 + 'A');
            }
            else 
            {
                printf("%c", plaintext[i]);
            }
        }
        printf("\n");
        
        return 0;
    }
    else 
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
}
